.pre-commit-config.yaml

```diff
@@ -15,22 +10,13 @@ repos:
           - id: check-yaml
           - id: debug-statements
             language_version: python3
-    - repo: https://github.com/PyCQA/flake8
-      rev: 7.0.0
-      hooks:
-          - id: flake8
-            language_version: python3
-            additional_dependencies: [flake8-typing-imports>=1.9.0]
-    - repo: https://github.com/asottile/reorder_python_imports
-      rev: v3.12.0
-      hooks:
-          - id: reorder-python-imports
-            args: ['--application-directories=.:src', --py3-plus]
-    - repo: https://github.com/asottile/pyupgrade
-      rev: v3.15.1
+    - repo: https://github.com/astral-sh/ruff-pre-commit
```

Comment:
Hopefully the final suggestion :)

Let's put the Ruff repo before the pre-commit-hooks one.

Reasoning: Ruff might do some formatting/fixes that prevent pre-commit-hooks failing.